### PR TITLE
Ticket #1946: Update field helper text

### LINKED ISSUE #1946: Django admin: Update helper text
### Issue description

Helper text appears below many of the fields in the admin. We audited this text across the admin and identified opportunities to make that info more meaningful and consistent.

### Acceptance criteria

- [ ] Update helper text within the admin as outlined within [this spreadsheet](https://docs.google.com/spreadsheets/d/1_4voSxjVVGnePbOdOBKV7Wdh830aZW5Tg8qsS59667Y/edit#gid=1452778825).

NOTE: Each admin page has a separate tab within that spreadsheet. In a [separate ticket](https://github.com/cisagov/manage.get.gov/issues/1920), we are updating field labels, so some of the labels identified in this spreadsheet may be outdated. But that should not affect the proposed changes to helper text.

### Additional context

_No response_

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1946 

## Changes

- Modifies admin.py to add custom help texts for is_superuser, is_staff, and password (as they are part of the base user class)
- Modifies help texts in accordance with [this doc](https://docs.google.com/spreadsheets/d/1_4voSxjVVGnePbOdOBKV7Wdh830aZW5Tg8qsS59667Y/edit#gid=999187084)
- Adds special state logic for Domain state (as it should differ based on what the current state is)
- Updated migrations
-  Adds [domain_fieldset.html](https://github.com/cisagov/manage.get.gov/pull/2032/files#diff-4008db232b0614d9feb558686464f55bd0470636c17c300d87ffd4e43f41de7a) for custom state messages. We will (likely) need this in the future anyway, its good we're adding this now.

## Context for reviewers
The meat and potatoes of this ticket resides [here](https://docs.google.com/spreadsheets/d/1_4voSxjVVGnePbOdOBKV7Wdh830aZW5Tg8qsS59667Y/edit#gid=999187084). In essence, this PR just replaces a bunch of help text fields. 
There are two edge cases that had to be worked around:
1. Domain state's help text changes based off of the given state
2. On user, the help text for password, is_staff, and is_superuser are all overridden. These reside on the base class so these had to be overwritten a bit differently

## Setup
All changes reside on /admin.
Go to the models and fields listed in the document, and compare their text content. All of them should be consistent with how it is outlined.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -48,6 +48,34 @@ class Meta:
...
+            is_superuser.help_text = "For development purposes only; provides superuser access on the database level."
+
+        if is_staff is not None:
+            is_staff.help_text = "Designates whether the user can log in to this admin site."
+
+        if password is not None:
+            # Link is copied from the base implementation of UserChangeForm.
+            link = f"../../{self.instance.pk}/password/"
+            password.help_text = (
+                "Raw passwords are not stored, so they will not display here. "
```
Comment (rachidatecs) [inline src/registrar/admin.py:75]: Wait a min. We're not using DJA passwords at all, are we? We're using login for auth - look at `user.set_unusable_password()` in oidc backends. Am I reading it right?

File: src/registrar/admin.py [hunk 2]
```diff
@@ -1662,20 +1690,26 @@ def changeform_view(self, request, object_id=None, form_url="", extra_context=No
         # Pass in what the an extended expiration date would be for the expiration date modal
         if object_id is not None:
             domain = Domain.objects.get(pk=object_id)
-            years_to_extend_by = self._get_calculated_years_for_exp_date(domain)
+            extra_context = self._set_expiration_date_context(domain, extra_context)
```
Comment (rachidatecs) [inline src/registrar/admin.py:1693]: Dude, you keep sneaking in revisions to this in unrelated PRs

File: src/registrar/models/domain.py [hunk 3]
```diff
@@ -992,20 +1018,22 @@ def __str__(self) -> str:
...
 
     state = FSMField(
         max_length=21,
         choices=State.choices,
         default=State.UNKNOWN,
-        protected=True,  # cannot change state directly, particularly in Django admin
-        help_text="Very basic info about the lifecycle of this domain object",
+        # cannot change state directly, particularly in Django admin
+        protected=True,
+        # This must be defined for custom state help messages,
```
Comment (rachidatecs) [inline src/registrar/models/domain.py:1029]: Nice comment

File: src/registrar/models/utility/generic_helper.py [hunk 4]
```diff
@@ -154,7 +154,7 @@ def _update_org_type_from_generic_org_and_election(self):
                 # There is no avenue for this to occur in the UI,
                 # as such - this can only occur if the object is initialized in this way.
                 # Or if there are pre-existing data.
-                logger.warning(
```
Comment (rachidatecs) [inline src/registrar/models/utility/generic_helper.py:157]: Nice

**Review by rachidatecs — COMMENTED** [review]
Initial pass, all looks good. I'll run the code and test more in round 2.

File: src/registrar/admin.py (hunk 1 above)
Comment (rachidatecs) [reply src/registrar/admin.py:75]: I suggest modifying the text to explain that password changes should be made on login.gov

**Review by rachidatecs — APPROVED** [review]
Figure out the password field and you're good to merge. 

I also suggest reuqesting a design review to go over every field's copy edits. If not, I'll give this a third pass this afternoon, let me know.

**Comment by Katherine-Osos** [comment]
@zandercymatics I sent you a few edits / questions in Slack.

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

**Pull request merged**